binding/binding.go

```diff
@@ -21,51 +27,26 @@ type Binder func(*http.Request, interface{}) error
 // CustomTypeDecoder converts a custom type from the request insto its exact type.
 type CustomTypeDecoder func([]string) (interface{}, error)
 
-// binders is a map of the defined content-type related binders.
-var binders = map[string]Binder{}
-
-var decoder *formam.Decoder
-var lock = &sync.Mutex{}
-var timeFormats = []string{
-	time.RFC3339,
-	"01/02/2006",
-	"2006-01-02",
-	"2006-01-02T15:04",
-	time.ANSIC,
-	time.UnixDate,
-	time.RubyDate,
-	time.RFC822,
-	time.RFC822Z,
-	time.RFC850,
-	time.RFC1123,
-	time.RFC1123Z,
-	time.RFC3339Nano,
-	time.Kitchen,
-	time.Stamp,
-	time.StampMilli,
-	time.StampMicro,
-	time.StampNano,
+// RequestTypeBinder are those capable of handling a request type like JSON or XML
+type RequestTypeBinder interface {
```

Comment:
This interface feels really, really backward to me. It's a strange level of indirection. As far as I can tell, that is on my struct and its my responsibility to call the `Register` method on the `RequestBinder`. Why?

There should be an interface for a binder.

Reply:
Thanks @markbates!

Reply:
Covered this one.